src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,42 @@
+package duke.command;
+
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * Command to delete a task with given index.
+ */
+public class DeleteCommand extends Command {
+
+    private int num;
+
+    /**
+     * Constructor for DeleteCommand.
+     *
+     * @param info Type of command
+     * @param num Index to delete
+     */
+    public DeleteCommand(String info, int num) {
+        super(info);
+        this.num = num;
+    }
+
+    public int getNum() {
+        return num;
+    }
+
+    /**
+     * Execute the delete command and remove the task at index.
```

Comment:
I like that you added descriptive JavaDoc comments for all of the public methods 👍. They are descriptive and make reading the code easy since it is clear what the methods are supposed to do.